sdks/python/apache_beam/examples/complete/game/game_stats.py

```diff
@@ -0,0 +1,262 @@
+recorded in the context of subsequent fixed windowing. (This could be used to
+tell us what games are giving us greater user retention).
+
+Run injector.Injector to generate pubsub data for this pipeline. The Injector
+documentation provides more detail on how to do this.
+
+The PubSub topic you specify should be the same topic to which the Injector is
+publishing.
+
+For a description of the usage and options, use -h or --help.
+
+Required options:
+  --output=OUTPUT_PATH
+
+Options with default values:
+  --input='gs://apache-beam-samples/game/gaming_data*.csv'
+  --team_window_duration=60
+  --allowed_lateness=120
+
+To specify a different runner:
+  --runner=YOUR_RUNNER
+
+NOTE: When specifying a different runner, additional runner-specific options
+      may have to be passed in as well
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+
+import argparse
+import csv
+import logging
+from datetime import datetime
+
+import apache_beam as beam
+from apache_beam.transforms import combiners
+
+
+def timestamp2str(t, fmt='%Y-%m-%d %H:%M:%S.000'):
+  """Converts a unix timestamp into a formatted string."""
+  return datetime.fromtimestamp(t).strftime(fmt)
+
+
+def parse_game_event(elem):
+  """Parses the raw game event info into a Python dictionary.
+
+  Each event line has the following format:
+    username,teamname,score,timestamp_in_ms,readable_time
+
+  e.g.:
+    user2_AsparagusPig,AsparagusPig,10,1445230923951,2015-11-02 09:09:28.224
+
+  The human-readable time string is not used here.
+  """
+  try:
+    row = list(csv.reader([elem]))[0]
+    yield {
+        'user': row[0],
+        'team': row[1],
+        'score': int(row[2]),
+        'timestamp': int(row[3]) / 1000.0,
+    }
+  except:  # pylint: disable=bare-except
+    logging.error('Parse error on "%s"', elem)
+
+
+class FormatTeamScores(beam.DoFn):
+  """Formats the data into the output format
+
+  Receives a (team, score) pair, extracts the window start timestamp, and
+  formats everything together into a string. Each string will be a line in the
+  output file.
+  """
+  def process(self, elem, window=beam.DoFn.WindowParam):
+    team, score = elem
+    window = timestamp2str(int(window.start))
+    yield 'window_start: %s, total_score: %s, team: %s' % (window, score, team)
+
+
+class CalculateSpammyUsers(beam.PTransform):
+  """Filter out all but those users with a high clickrate, which we will
+  consider as 'spammy' uesrs.
+
+  We do this by finding the mean total score per user, then using that
+  information as a side input to filter out all but those user scores that are
+  larger than (mean * SCORE_WEIGHT).
+  """
+  SCORE_WEIGHT = 2.5
+
+  def expand(self, user_scores):
+    sum_scores = (
+        user_scores
+        | 'SumUsersScores' >> beam.CombinePerKey(sum)
+    )
+
+    global_mean_score = (
+        sum_scores
+        | beam.Values()
+        | beam.CombineGlobally(combiners.MeanCombineFn()).as_singleton_view()
```

Comment:
combiners are not supported in DataflowRunner streaming. You can just add a note, or change to GBK | Map and add a TODO to change it later.